### PR TITLE
fix: correctly sort compatible versions by major, then minor, and otherwise by locale

### PR DESCRIPTION
**Pull request by vladfrangu**
Should fix the CI bug found in #78 🎉

### CODE REVIEW

**Comment by mnmkng** [comment]
But it does not seem to have fixed the CI issue :D Or would it only apply after we merge it?

**Comment by vladfrangu** [comment]
> But it does not seem to have fixed the CI issue :D Or would it only apply after we merge it?

I need to change the way we find the versions from the website to check for the major first and then ensure the minor is >= expected minor (which is the third number in the version...Thanks google)

File: node-puppeteer-chrome/puppeteer_download.js [hunk 1]
```diff
@@ -15,7 +15,14 @@ async function downloadLatestCompatibleChrome() {
...
     const compatibleChromeVersion = matchedCompatibilityVersions
         .map((v) => v.chrome)
-        .sort((a, b) => b.localeCompare(a));
+        .sort((a, b) => {
+            const [majorA, minorA] = a.split('.').map((v) => Number(v));
+            const [majorB, minorB] = b.split('.').map((v) => Number(v));
+
+            if (majorB > majorA) return 1;
+            if (minorB > minorA) return 1;
+            return b.localeCompare(a);
```
Comment (B4nan) [inline node-puppeteer-chrome/puppeteer_download.js:24]: isnt this wrong? i mean if you compare a=3.2.1 vs b=3.2.10, both conditions are false and you end up doing local compare which will fail to sort this, or not?

might be safer to build a number from the version string (e.g. patch + 100 * minor + 10000 * major) and compare those

**Comment by vladfrangu** [comment]
-.-" Puppeteer 14 wants a version of Chrome that isn't promoted to google-chrome-stable yet...

**Comment by B4nan** [comment]
ideally we should still respect that version, maybe we could get it via http://unpkg.com/browse/apify@latest/package.json?

**Comment by vladfrangu** [comment]
> why is it using puppeteer v14? i though it was bound to the version from SDK, i guess that does not apply to the CI workflow, right?

CI tests with beta of apify and latest of everything else. Usually this works, except now when Puppeteer did this again...

**Comment by vladfrangu** [comment]
Come to think of it, the test shouldn't even fail, puppeteer will still be compatible with the older version of Chrome, right?

**Comment by B4nan** [comment]
God knows, should be, but I believe the table we use to get versions from states the version it was tested on, so probably no guaranteers for other versions.

My question is why only the node 14 test fails, if its an issue with downloading correct chrome version, how come node 15 and 16 are passing? :D

**Comment by B4nan** [comment]
oh wait a sec, SDK does not support node 14, i guess there will be some engines check that alters the resolution. 

i guess we can remove node 14 from the matrix here, right @mnmkng?

**Comment by mnmkng** [comment]
Yeah, Let's remove Node 14. Might wanna replace it with 18

**Comment by vladfrangu** [comment]
Lets see how nodejs 18 works 😄 🎉

**Comment by B4nan** [comment]
Lets remove node 14 now and merge this one. 

Then we can proceed with fixing sqlite3 prebuilds (I already tried to upgrade that some time ago and failed, so its like halfway done but some issues to be resolved) and adding node 18 afterwards. But before doing that I'd like to see the memory storage being merged (so the e2e tests needs to be fixed if they are not yet).

**Pull request closed without merging**